### ISSUE TITLE
MA0099 is triggered for quick HasFlag check

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#583]
### Version of the Meziantou.Analyzer NuGet package

2.0.74

### Rule Identifier

MA0099

### Target Framework

net8.0

### C# Language version

C# 12

### Description

`MA0099` is reported when checking that a partucular enum value have some flag. I can indeed use `HasFlag` check, but it's slow and requestes boxing.

### Reproduction Steps

Minimal code:

```c#
private static bool CanUpdate(ChannelPermissions permissions) =>
    (permissions.ChannelMetadataPermissions & ChannelMetadataPermissions.Update) != 0; // <- use enum value instead of 0
```

**Comment by meziantou** [comment]
You can use `ChannelMetadataPermissions.None` if it exists or explicitly cast the value `(ChannelMetadataPermissions)0`. If you find it clear to use the implicit conversion, you can disable the rule.

> I can indeed use HasFlag check, but it's slow and requests boxing.

This is not the case on .NET 5+. The JIT is able to optimize the call.
